source/extensions/filters/http/grpc_web/grpc_web_filter.cc

```diff
@@ -156,6 +180,12 @@ Http::FilterHeadersStatus GrpcWebFilter::encodeHeaders(Http::ResponseHeaderMap&
   } else {
     headers.setReferenceContentType(Http::Headers::get().ContentTypeValues.GrpcWebProto);
   }
+
+  if (!checkGrpcResponseHeaders(headers, end_stream)) {
```

Comment:
Is stopping iteration correct when end_stream=true? We wouldn't get a subsequent data frame that could be used to continue iteration